workflow/controller/controller.go

```diff
@@ -512,7 +512,18 @@ func (wfc *WorkflowController) processNextPodCleanupItem(ctx context.Context) bo
 		pods := wfc.kubeclientset.CoreV1().Pods(namespace)
 		switch action {
 		case terminateContainers:
-			if terminationGracePeriod, err := wfc.signalContainers(namespace, podName, syscall.SIGTERM); err != nil {
+			pod, err := wfc.getPod(namespace, podName)
+			if err == nil && pod != nil && pod.Status.Phase == "Pending" {
+				logCtx.Info("deleting pending pod")
```

Comment:
Please could you add `.WithFields(log.Fields{"name": podName, "namespace": namespace})` to the logging.

Reply:
ok，I will add it.